goss-testing/scripts/check_bios_firmware_versions.sh

```diff
@@ -417,26 +440,45 @@ check_if_bmcs_are_reachable
 # for each BMC
 for i in "${NCN_BMCS[@]}"
 do 
-  
+
+  if [[ "$VENDOR" = *"Marvell"* ]] \
+    || [[ "$VENDOR" = "HP"* ]] \
+    || [[ "$VENDOR" = "Hewlett"* ]]; then
+    if [[ "$i" == *"ncn-m001"* ]]; then
+      # login locally
+      ilorest --nologo login 1>/dev/null
+    else
+      # login to the remote bmc 
+      ilorest --nologo login "$i" 1>/dev/null
+    fi
+
+  fi
+
   check_firmware_version "$i" \
     "$BMC_USERNAME" \
     "$IPMI_PASSWORD"
 
   check_bios_version "$i" \
     "$BMC_USERNAME" \
     "$IPMI_PASSWORD"
-    
+
+  if [[ "$VENDOR" = *"Marvell"* ]] \
+    || [[ "$VENDOR" = "HP"* ]] \
+    || [[ "$VENDOR" = "Hewlett"* ]]; then
+    ilorest --nologo logout 1>/dev/null
+  fi
+
 done
 
 if [[ $BASELINE = Y ]]; then
-  /root/bin/bios-baseline.sh --check
+  /root/bin/bios-baseline.sh --check || rc=1
   exit $?
```

Comment:
You also need to delete this line, or the change on the previous line will make things even worse.

```suggestion
```

Reply:
Fixed

Reply:
You're right, I deleted the wrong `exit`.  Fixed in this last push